python/pyarrow/_parquet.pyx

```diff
@@ -536,6 +536,37 @@ cdef class FileMetaData:
             c_string c_path = tobytes(path)
         self._metadata.set_file_path(c_path)
 
+    def append_row_groups(self, FileMetaData other):
+        """
+        Append row groups of other FileMetaData object
+        """
+        cdef shared_ptr[CFileMetaData] c_metadata
+
+        c_metadata = other.sp_metadata
+        self._metadata.AppendRowGroups(c_metadata)
```

Comment:
More reason to remove the possibility of exception being thrown in this function -- it's not being caught here